python_modules/libraries/dagster-k8s/dagster_k8s/pipes.py

```diff
@@ -1,12 +1,17 @@
+import logging
 import os
 import random
 import re
 import string
+import threading
+import time
+from collections.abc import Callable, Generator
 from contextlib import contextmanager
 from pathlib import Path
 from typing import Any, Iterator, Mapping, Optional, Sequence, Set, Union
 
 import kubernetes
+import pendulum
```

Comment:
`pendulum` isn't part of current dependencies, can we replace it with `datetime`? 

Currently CI is failing because of this issue.

Reply:
Hey @danielgafni, addressed this change though it's a bit janky due to how datetimes work in Python <3.10

Let me know how it looks. I tested in Python 3.10 and 3.11 and both tests 
![image](https://github.com/user-attachments/assets/7ab76d39-4ee1-4577-9118-692cf607da60)